bedwars-plugin/src/main/java/com/andrei1058/bedwars/arena/Arena.java

```diff
@@ -714,6 +715,13 @@ public boolean addSpectator(@NotNull Player p, boolean playerBefore, Location st
 
             p.sendMessage(getMsg(p, Messages.COMMAND_JOIN_SPECTATOR_MSG).replace("{arena}", this.getDisplayName()));
 
+            ArenaSpectateEvent spectateEvent = new ArenaSpectateEvent(p, this);
+            Bukkit.getPluginManager().callEvent(spectateEvent);
+            if (spectateEvent.isCancelled()) {
+                spectateEvent.setCancelled(true);
+                return true;
+            }
+
```

Comment:
Lets say the spacetateEvent is canceled, what happens then? as the player already joined the arena.

Reply:
> Lets say the spacetateEvent is canceled, what happens then? as the player already joined the arena.

During game play, game players cannot spectate the game. If the game has already started and players wish to spectate, they can spectate. If you cancel this event, you will not be able to spectate.

Reply:
No thats not how it works, Because you added the specatate event at the bottom of the function it will only cancel anything thats behind the event call...

Reply:
> No thats not how it works, Because you added the specatate event at the bottom of the function it will only cancel anything thats behind the event call...

Changed code location to make ArenaSpectateEvent work properly